### ISSUE TITLE
Question about customize objective function

### ISSUE THREAD

**Issue by ajing** [dmlc/xgboost#407]
Could I ask a trivial question? For the example provided in custom_objective.R, the first gradient is 1/(1 + exp(-preds)) - labels. But the function for log loss is -log P(yt|yp) = -(yt log(yp) + (1 - yt) log(1 - yp)). Is first gradient the derivative for log loss function? I have an object function y = f(x), which I don't know the form of f(). I can get y by input x. Could I do something like (f(x+dx) - f(x-dx))/(2*dx) to get the first gradient? Thanks for the help!

**Comment by tqchen** [comment]
assume logloss equals f(yt, yp), what we want is partial_yp f(yt, yp), which is the given value.
I think numerical gradient (f(x+dx) - f(x-dx))/(2*dx) might be valid, but not accurate.
